core/iwasm/compilation/aot_llvm.c

```diff
@@ -2782,6 +2782,10 @@ aot_create_comp_context(const AOTCompData *comp_data, aot_comp_option_t option)
                 bh_assert(vendor_sys);
                 bh_memcpy_s(default_arch, sizeof(default_arch), default_triple,
                             (uint32)(vendor_sys - default_triple));
+                if (!strcmp(default_arch, "arm64")) {
+                    bh_strcpy_s(default_arch, sizeof(default_arch),
+                                "aarch64v8");
```

Comment:
And this makes arm64 a supported target for wamrc, wondering is it better to the target list of `wamrc --help`?
```bash
Usage: wamrc [options] -o output_file wasm_file
  --target=<arch-name>      Set the target arch, which has the general format: <arch><sub>
                            <arch> = x86_64, i386, aarch64, arm, thumb, xtensa, mips,
                                     riscv64, riscv32.
                              Default is host arch, e.g. x86_64
```

Reply:
> Not sure why modify the code, seems it doesn't take effect when `wamrc --target=arm64 -o ? wamrc reported the error below:
> 
> ```shell
> Error: Invalid target. Use --target=help to list all supported targets
> ```
> 
> How about:
> 
> ```diff
> diff --git a/core/iwasm/compilation/aot_llvm.c b/core/iwasm/compilation/aot_llvm.c
> index 2e1e82e5..6f7a7d56 100644
> --- a/core/iwasm/compilation/aot_llvm.c
> +++ b/core/iwasm/compilation/aot_llvm.c
> @@ -2712,7 +2712,7 @@ aot_create_comp_context(const AOTCompData *comp_data, aot_comp_option_t option)
>                  arch = "thumbv4t";
>              else if (!strcmp(arch, "thumbeb"))
>                  arch = "thumbv4teb";
> -            else if (!strcmp(arch, "aarch64"))
> +            else if (!strcmp(arch, "aarch64") || !strcmp(arch, "arm64"))
>                  arch = "aarch64v8";
>              else if (!strcmp(arch, "aarch64_be"))
>                  arch = "aarch64_bev8";
> ```

arm64 never was a supported target before though and since it really appears to just be an alias for aarch64v8 does it make sense to introduce it as one now?

Reply:
> Not sure why modify the code, seems it doesn't take effect when `wamrc --target=arm64 -o ? wamrc reported the error below:
> 

The intention of the patch wasn't really to introduce arm64 as a valid target but rather to convert it aarch64v8 in the one place where one could set it (by not passing a target when running wamrc on a mac).

Reply:
OK, if that, seems had better not introduce arm64 as a valid target. IIUC, you compile wamrc and use it in mac arm64, and here the gotten default_arch is `arm64` and you want to normalize it into `aarch64v8`?

Reply:
how about adding a comment here?

Reply:
> OK, if that, seems had better not introduce arm64 as a valid target. IIUC, you compile wamrc and use it in mac arm64, and here the gotten default_arch is `arm64` and you want to normalize it into `aarch64v8`?

That's correct.